File: app/src/main/java/net/hearnsoft/gensokyoradio/trd/ws/GRWebSocketClient.java
package net.hearnsoft.gensokyoradio.trd.ws;

import android.util.Log;

import org.java_websocket.client.WebSocketClient;
import org.java_websocket.drafts.Draft_6455;
import org.java_websocket.handshake.ServerHandshake;

import java.net.URI;

import javax.net.ssl.SSLParameters;

public class GRWebSocketClient extends WebSocketClient {

    private static final String TAG = GRWebSocketClient.class.getSimpleName();

    public GRWebSocketClient(URI serverUri) {
        super(serverUri, new Draft_6455());
        Log.d(TAG, "Init WebSocket client.");
    }

    @Override
    protected void onSetSSLParameters(SSLParameters sslParameters) {
        //super.onSetSSLParameters(sslParameters);
    }

    @Override
    public void onOpen(ServerHandshake handShakeData) {
        Log.d(TAG, "Connection opened. get handshake: " + handShakeData.getHttpStatusMessage());
    }

    @Override
    public void onMessage(String message) {
        Log.d(TAG, "onMessage: " + message);
    }

    @Override
    public void onClose(int code, String reason, boolean remote) {
        Log.d(TAG, "onClose: code:" + code + ", reason:" + reason + ", isRemoteClose:" + remote);
    }

    @Override
    public void onError(Exception ex) {
        Log.e(TAG, "onError: " + ex.toString());
        ex.printStackTrace();
    }
}

File: app/src/main/java/net/hearnsoft/gr3rd/compose/socket/GRWebSocketClient.kt
package net.hearnsoft.gr3rd.compose.socket

import net.hearnsoft.gr3rd.compose.utils.Logger
import org.java_websocket.client.WebSocketClient
import org.java_websocket.drafts.Draft_6455
import org.java_websocket.handshake.ServerHandshake
import java.net.URI
import javax.net.ssl.SSLParameters

class GRWebSocketClient(
    serverUri: URI,
    private val onMessage: (String) -> Unit = {},
    private val onClose: (Int, String, Boolean) -> Unit = { _, _, _ -> },
    private val onOpen: (ServerHandshake) -> Unit = {},
    private val onError: (Exception) -> Unit = {}
) : WebSocketClient(serverUri, Draft_6455()) {
    override fun onSetSSLParameters(sslParameters: SSLParameters?) {

    }

    override fun onOpen(handshake: ServerHandshake) {
        Logger.info(this, "WebSocket connection opened: $handshake")
        onOpen.invoke(handshake)
    }

    override fun onMessage(message: String) {
        Logger.info(this, "Received message: $message")
        onMessage.invoke(message)
    }

    override fun onClose(code: Int, reason: String, remote: Boolean) {
        Logger.info(this, "WebSocket connection closed: code=$code, reason=$reason, remote=$remote")
        onClose.invoke(code, reason, remote)
    }

    override fun onError(ex: Exception) {
        Logger.err(this, "WebSocket error occurred", ex)
        onError.invoke(ex)
    }

}